src/client/gui.js

```diff
@@ -1,8 +1,236 @@
-/* globals ProjectDialogMorph, ensureFullUrl, Morph, AlignmentMorph, InputFieldMorph, localize,
+/* globals ProjectDialogMorph, world, SymbolMorph, PushButtonMorph, ToggleButtonMorph, ensureFullUrl, Morph, AlignmentMorph, InputFieldMorph, localize,
    Point, TextMorph, Color, nop, ListMorph, IDE_Morph, Process, BlockImportDialogMorph,
    BlockExportDialogMorph, detect, SnapCloud, SnapSerializer, ScrollFrameMorph,
    DialogBoxMorph, SnapActions, SpeechBubbleMorph
    */
+
+function isMobileDevice() {
+    return (typeof window.orientation !== 'undefined') || (navigator.userAgent.indexOf('Mobile') !== -1);
+}
+
+var _toggleAppMode = IDE_Morph.prototype.toggleAppMode;
+IDE_Morph.prototype.toggleAppMode = function() {
+    var ideMorphIns = world.children[0];
+    _toggleAppMode.apply(ideMorphIns, arguments);
+    if (isMobileDevice()) {
+        mobileMode.init();
+    }
+};
+
+var _ideFixLayout = IDE_Morph.prototype.fixLayout;
+IDE_Morph.prototype.fixLayout = function() {
+    var ideMorphIns = world.children[0];
```

Comment:
why not `this`?

Reply:
it would probably be better to store the old `fixLayout` method on the prototype still (rather than a new variable).